### DISCUSSION TITLE
L0 checkpointing network as IPC rootnet based on Braid

### DISCUSSION THREAD

**Discussion by adlrocha** [consensus-shipyard/consensuslab#166] [Project proposals]
There have been discussions around what a good rootnet for IPC (former Hierarchical Consensus protocol) would look like. In one of our CoLos we discussed the possibility of using as a rootnet a specialized L0 that only supports the commitment of checkpoints from subnets, and the propagation of messages included in these checkpoints (without any general-purpose smart contract support). This L0 could be based on [Braid](https://github.com/protocol/braid/blob/master/DESIGNDOC.md) and be also checkpointing in parallel to other major chains in the ecosystem (BTC, ETCH, etc.).

Two of the key design decisions for this L0 network is what consensus algorithm to use (PoW, PoS, PBFT-based, etc.), and what are the incentives for nodes participating and securing the network.

**Comment by aakoshh** [comment]
The Braid link doesn't work for me. 

One question is how to allow hitherto unknown consensus algorithms to checkpoint themselves to the rootnet, without giving them the ability to deploy a subnet actor that knows how to validate their checkpoints. Would it be some governance process on the rootnet that vets actor code to check that it's really checkpointing related?

**Reply by jsoares** [reply]
Same here. We may need a brief explanation that doesn't rely on the repo.

**Comment by vukolic** [comment]
Sybil attack protection (and consequently security) for this network is key here. 

In principle, in case of a bad choice of Sybil protection, this is a possible centralization point and networks might opt to directly anchor into L1's (e.g., Bitcoin, Filecoin) instead to this one to avoid a level of indirection.
